.github/workflows/go-standards.yml

```
@@ -0,0 +1,62 @@
+# This workflow will build, test, and lint a Golang project.
+# For more information see:
+# - https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+# - https://golangci-lint.run/usage/install/#github-action
+
+name: Go Standards (Build, Test & Lint)
+
+on:
+  push:
+    branches: ["main", "ci", "test"]
+  pull_request:
+    branches: ["main", "ci", "test"]
+
+permissions:
+  contents: read
+
+jobs:
+  build-test-lint:
+    # This job runs on the latest Ubuntu environment
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Check out the repository
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Step 2: Set up Go environment
+      - name: Set up Go v1.22.3
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      # Step 3: Build the project
+      - name: Build the project
+        run: go build -v ./...
+        # Ensures build step is only attempted if the repository was checked out successfully
+        continue-on-error: false
+
+      # Step 4: Test the project
+      - name: Run tests
+        run: go test -v ./... -test.short
```

```suggestion
        run: go test -v ./...
```

I'm unsure you should only run the short ones.